readme_renderer/rst.py

```diff
@@ -129,4 +133,17 @@ def render(
     if rendered:
         return clean(rendered)
     else:
+        # If the warnings stream is empty, docutils had none, so add ours.
+
+        # We have to currently handle the legacy disutils integration check
+        # command, as it implements its own warnings stream structure.
+        # Once that code is removed, we can remove this conditional.
+        if hasattr(stream, "output"):
+            if not stream.output.getvalue():  # type: ignore[union-attr]
+                stream.write("No content rendered from RST source.")
```

Comment:
@di The `check` command was deprecated back in https://github.com/pypa/readme_renderer/pull/122 - are we fine with removing that code? It'd make this code a little cleaner if it didn't have to consider that command.

Reply:
The answer here might make this moot, but looking at the code for `Check`:

https://github.com/pypa/readme_renderer/blob/361f7fc598c9333f6ba3755e9f80e171d0bdce5a/readme_renderer/integration/distutils.py#L97-L105

It seems like it could be reworked to remove `_WarningStream`, and pass in `io.StringIO` instead. That would simplify this conditional, and remove the circular import.

Reply:
Although it looks like Twine would need to make a similar change (which I’m happy to do):

https://github.com/pypa/twine/blob/a6dd69c79f7b5abfb79022092a5d3776a499e31b/twine/commands/check.py#L51

Reply:
Instead of reworking this deprecated code, I've prepared #233 to remove it, and if that gets merged, I can simplify this.

Reply:
Nice to see it removed, but to be clear, not checking for `stream.output` means Twine will also need to replace the `_WarningStream` class. I'm happy to do that, but it'll be this weekend at the earliest. It'd be nice if we held off on releasing `readme_renderer` until that's done, to avoid breaking current workflows.

Reply:
@bhrutledge no worries, makes sense! This code isn't ready to merge yet - I'm updating it now to account for the removal, and will convert to Draft to make sure someone doesn't accidentally merge until twine is updated.

Reply:
Twine 4.0.0 is released, which should support this change.